src/ec/packages/ec-component-date-block/ecl-date-block.html.twig

```diff
@@ -41,32 +41,38 @@
 {% if extra_attributes is defined and extra_attributes is not empty and extra_attributes is iterable %}
   {% for attr in extra_attributes %}
     {% if attr.value is defined %}
-      {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value|e('html_attr') ~ '"' %}
+      {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name|e('html_attr') ~ '="' ~ attr.value|e('html_attr') ~ '"' %}
     {% else %}
-      {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name %}
+      {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name|e('html_attr') %}
     {% endif %}
   {% endfor %}
 {% endif %}
 
-{% if (_month is not empty or _month_full is not empty) and _year is not empty %}
+{% if _date_time is not empty %}
+  {% set _extra_attributes = _extra_attributes ~ 'dateTime="' ~ _date_time|e('html_attr') ~ '"' %}
+{% endif %}
+
+<time class="{{ _css_class }}"{{ _extra_attributes|raw }}>
+  {% if _date_time is not empty %}
+    <span class="ecl-u-sr-only">{{ _date_time }}</span>
+  {% endif %}
+  <span class="ecl-date-block__day" aria-hidden="true">{{ _day }}</span>
+{% if (_month is not empty or _month_full is not empty) %}
   {% if _month_full is not empty and _month_full != _month %}
-    {% set _month_markup = '<abbr title="' ~ _month_full ~ '" class="ecl-date-block__month" aria-hidden="true">' ~ _month ~ '</abbr>' %}
+  <abbr
+    title="{{ _month_full|e('html_attr') }}"
```

Comment:
These changes reminded me that looks like have to do that in other components. I saw in other components we leave construction like that: `attrname="{{ _variable }}"` with relying on enabled autoescape in twig.
But problem is that autoescape in twig uses HTML strategy (https://twig.symfony.com/doc/1.x/tags/autoescape.html ). So it will be not equal to html_attr.
By following a more strict approach and consistency we have to do the same in other components.
What do you think?

Reply:
I agree that if we do these checks we have to do them consistently. I will browse again the library to find other use cases of attribute's values not being escaped, the idea is indeed to have an overall approach and use it consistently, enforcing "security" on attributes using the right escaping strategy from now on.

Reply:
We had a discussion today with @ademarco and in the context of changes related to security updates, discovered that last changes not all of them fully in line with OWASP recommendations. For instance according to documentation of twig: https://twig.symfony.com/doc/3.x/filters/escape.html and  https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.md#rule-2---attribute-encode-before-inserting-untrusted-data-into-html-common-attributes we shouldn't use everywhere only html_attr. Looks like for some special attributes we have to use strategy according to the purpose of the attribute.
As I understand for `src` and `href` we have to use `url` strategy and for `style` - `css`.

Reply:
makes sense, will spot these specific use cases, thanks for reporting it ;)

Reply:
ummm...no, the docs on twig say that `url` strategy should not be used for full urls but only for segments, it escapes the slashes, so even a path would be broken, the hashes too, so that anchor links would be broken... My feeling is that we can stick to `html_attr` as a stronger strategy than the default html, and that's it..

Reply:
So...we started having issues on our side as soon as we extended the scope of this pull request, basically to apply a stricter escaping strategy on all the attributes.
Therefore i decided to revert the latest changes and bring back this pull request to its former scope, fixing the double escaping introduced by the removal of the `|raw` filter that was breaking the js and the attribute's functionality in general.

I think this is the safest thing to do, it all originated from the security issues related to the use of `|raw`, we now only use it for strings that we have hardcoded in the templates or that we constructed with parameters that we always previously escaped with `e('html_attr')`. 

All the other use cases of attributes that we set in html in the template and where only their value comes from a variable are untouched, those don't need the usage of `|raw` to avoid double escaping, indeed.

Reply:
I suppose it is fine.